sklearnex/_device_offload.py

```diff
@@ -107,10 +107,9 @@ def _get_global_queue():
 
 
 def _transfer_to_host(queue, *data):
-    has_usm_data, has_host_data = False, False
-
     host_data = []
     for item in data:
+        has_usm_data, has_host_data = False, False
```

Comment:
This change is a bit surprising. @Alexsandruss @samir-nasibli what do you think? I want more clarification on this change because it removes hysteresis and blocks checks later on in the function

Reply:
it obviously was a bug. this flags should be independent for each item in data.

Reply:
it generates a fail on current master branch if first item in data has usm interface and the second one doesn't (it is exactly like in IncPCA.partial_fit. probably the right way was just delete the this check because it looks like it has been designed exactly to fail in such situation (have no idea why it is necessary)

Reply:
I think that is exactly what it was meant to check for. check_input should be passed into dispatch as a kwargs (i.e. check_input=check_input)

Reply:
oh, you are right, I missed this way. thanks for pointing out.

Reply:
please remove blocking status

Reply:
It looks as though the changes haven't been fully reverted, could you please move ```has_usm_data, has_host_data = False, False``` back to the original spot? If its still up for debate, I would recommend putting it into a separate PR.

Reply:
Sorry, github was causing the versions to be odd. I will ask for the changes requested to be removed.